### ISSUE TITLE
🐞 v1 - Unexpected behaviours occur when no abnormal images provided in the validation set

### ISSUE THREAD

**Issue by samet-akcay** [openvinotoolkit/anomalib#1599]
### Describe the bug

This is an edge case, where the validation set only contains normal images. Training a PADIM model on the hazelnut toy dataset returns 0% performance.

### Dataset

Other (please specify in the text field below)

### Model

PADiM

### Steps to reproduce the behavior

```bash
anomalib train --data normal.yaml --model Padim --task CLASSIFICATION
```

Using the below configuration as `normal.yaml`, the above CLI command will return 0% performance.

### OS information

OS information:
- OS: Manjaro
- Python version: 3.11
- Anomalib version: v1.0.0
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: 12
- GPU models and configuration: 2x GeForce RTX 3090
- Any other relevant information: [e.g. I'm hazelnut_toy dataset]

### Expected behavior

- Train the model.
- Normalize the scores [optional]
- Find the adaptive threshold, or use the provided manual threshold.

### Screenshots

![image](https://github.com/openvinotoolkit/anomalib/assets/17574420/ebf175be-0d15-4b9f-a87f-e8ea2c8ec266)

### Pip/GitHub

pip

### What version/branch did you use?

v1.0.0

### Configuration YAML

```yaml
class_path: anomalib.data.Folder
init_args:
  root: "datasets/hazelnut_toy"
  normal_dir: "good"
  normalization: imagenet
```

### Logs

```shell
Testing DataLoader 0: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  2.50it/s]
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │            0.0            │
│       image_F1Score       │            0.0            │
└───────────────────────────┴───────────────────────────┘
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
I think I have encountered a similar thing before. This is a problem of metrics if only a single value is present. I sadly don't know if I managed to fix this.

**Comment by samet-akcay** [comment]
@blaz-r, you are right. It turned out that it is 0% because there was no anomalous images in the dataset. `torchvision` shrank all the values to 0. There is a warning message, but is well hidden in the clutter. 

We need to de-clutter the logs to avoid this.
